client/jetpack-connect/signup.js

```diff
@@ -83,11 +95,47 @@ export class JetpackSignup extends Component {
 		} );
 	}
 
+	componentDidUpdate( prevProps ) {
+		const { requestError } = this.props;
+
+		if ( prevProps.requestError || ! requestError ) {
+			return;
+		}
+
+		if (
+			this.getWooDnaConfig() &&
+			'usernameOrEmail' === requestError.field &&
+			'unknown_user' === requestError.code
+		) {
+			this.showWooDnaSignupView();
+		}
+	}
+
+	showWooDnaSignupView = () => {
+		this.setState( {
+			showWooDnaLoginForm: false,
+		} );
+		this.props.resetAuthAccountType();
+	};
+
+	showWooDnaLoginView = ( usernameOrEmail ) => {
+		this.setState( {
+			showWooDnaLoginForm: true,
+			signUpUsernameOrEmail: usernameOrEmail || null,
+		} );
+		this.props.resetAuthAccountType();
```

Comment:
What does `resetAuthAccountType` do in this case?

Reply:
When you put an email/username in the initial "login" form, a request is made to the server to see what kind of account that username/email has. The possible answers can be:
- Doesn't exist
- Regular account
- Passwordless account

Depending on that answer, the next step of the form will be different.
- If there is no account, the signup form will render.
- If it's a regular account, the password field will appear.
- If it's a passwordless account, a notice telling the user to check their e-mail will appear (I haven't tested this one out yet).

In this case, because the user is going back to the initial "login" form, we want to clear that response so they can input another username/email.